### ISSUE TITLE
Add the rule `use-resource-hints`

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#287]
- URL duplication checking
- Switch whether it requires `<meta http-equiv='x-dns-prefetch-control' content='on'>`
- `prerender`: checking whether used only once.
- Switch whether it requires the `pr` attribute.
- Switch whether using the `preload`.

**Comment by YusukeHirao** [comment]
Closing as not planned. This rule was proposed over 4 years ago for v3.x and the web platform has evolved since then (e.g., `prerender` is being superseded by the Speculation Rules API). The implementation complexity is high with 5 sub-checks, and there has been no community demand.

If there is renewed interest in resource hints linting, a new issue with updated requirements would be welcome.
